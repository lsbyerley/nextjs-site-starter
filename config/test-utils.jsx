// test-utils.js
import { render } from '@testing-library/react';

// Add in any providers here if necessary:
// (ReduxProvider, ThemeProvider, etc)
const Providers = ({ children }) => {
  return children;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export {
  default as userEvent,
  specialChars,
} from '@testing-library/user-event';
// export * from 'next-page-tester';

// override render method
export { customRender as render };
