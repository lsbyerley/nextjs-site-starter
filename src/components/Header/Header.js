import Link from 'next/link';
import React from 'react';
import ThemeSwitch from '@/components/ThemeSwitch';

const Header = () => {
  return (
    <div className='sticky top-0 z-20 py-2 bg-white md:py-6 md:mb-6 dark:bg-black'>
      <div className='container flex items-center justify-between px-4 mx-auto lg:max-w-4xl'>
        <Link href='/'>
          <a
            className={
              'font-medium tracking-wider transition-colors text-gray-900 hover:text-sky-500 uppercase dark:text-white'
            }
          >
            Your Name
          </a>
        </Link>
        <ThemeSwitch />
      </div>
    </div>
  );
};

export default Header;
