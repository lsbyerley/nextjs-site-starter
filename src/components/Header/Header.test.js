import React from 'react';
import { render, screen } from 'test-utils';
import Header from './Header';

jest.mock('@/components/ThemeSwitch', () => ({
  __esModule: true,
  default: jest.fn().mockImplementation((props) => {
    return <div role='switch'></div>;
  }),
}));

const renderHeader = (props) => {
  const utils = render(<Header {...props} />);
  const header = utils.container.firstChild;
  return { ...utils, header };
};

describe('Header', () => {
  it('renders and matches snapshot', () => {
    const { header } = renderHeader();
    expect(header).toBeTruthy();
    expect(header).toMatchSnapshot();
  });
  it('shows Your Name intro', () => {
    renderHeader();
    const yourName = screen.getByText('Your Name');
    expect(yourName).toBeInTheDocument();
  });
});
