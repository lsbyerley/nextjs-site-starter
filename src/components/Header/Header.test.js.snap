// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders and matches snapshot 1`] = `
<div
  class="sticky top-0 z-20 py-2 bg-white md:py-6 md:mb-6 dark:bg-black"
>
  <div
    class="container flex items-center justify-between px-4 mx-auto lg:max-w-4xl"
  >
    <a
      class="font-medium tracking-wider transition-colors text-gray-900 hover:text-sky-500 uppercase dark:text-white"
      href="/"
    >
      Your Name
    </a>
    <div
      role="switch"
    />
  </div>
</div>
`;
